mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h

```diff
@@ -108,15 +109,23 @@ struct ConvolutionContext : SQLiteSerializable<ConvolutionContext> {
 // T is either GriwiseGemmOp or GridwiseGemmV2Op
 template <typename T> static miopen::ConvOpType ObtainConvDirection(T &op) {
   miopen::ConvOpType opType;
-  auto kernel_algorithm =
-      op->template getAttrOfType<StringAttr>("kernel_algorithm").getValue();
-  if (kernel_algorithm.find(StringRef("backward_data")) != StringRef::npos) {
+  if (isa<miopen::Conv2DOp>(*op)) {
+    opType = miopen::ConvOpType::Conv2DOpType;
+  } else if (isa<miopen::Conv2DBwdDataOp>(*op)) {
     opType = miopen::ConvOpType::Conv2DBwdDataOpType;
-  } else if (kernel_algorithm.find(StringRef("backward_weight")) !=
-             StringRef::npos) {
+  } else if (isa<miopen::Conv2DBwdWeightOp>(*op)) {
     opType = miopen::ConvOpType::Conv2DBwdWeightOpType;
   } else {
-    opType = miopen::ConvOpType::Conv2DOpType;
+    auto kernel_algorithm =
+        op->template getAttrOfType<StringAttr>("kernel_algorithm").getValue();
+    if (kernel_algorithm.find(StringRef("backward_data")) != StringRef::npos) {
+      opType = miopen::ConvOpType::Conv2DBwdDataOpType;
+    } else if (kernel_algorithm.find(StringRef("backward_weight")) !=
+               StringRef::npos) {
+      opType = miopen::ConvOpType::Conv2DBwdWeightOpType;
+    } else {
+      opType = miopen::ConvOpType::Conv2DOpType;
+    }
```

Comment:
Under what circumstance would this chunk of code be executed? Looks like at this point it isn't either fwd/bwd or wrw.

Reply:
I haven't finished changing the unit tests yet so the current implementation allows `miopen-affix-params` be the 1st OR the 3rd pass, hence this logic.

In the next iteration I'll eliminate these logic so `miopen-affix-params` must be the 1st pass in the pipeline.